### PR TITLE
Allows pet owners to name and describe their pet

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title.

## How This Contributes To The Skyrat Roleplay Experience

Much like how loadout items can be renamed, this should proably be a thing too.

Pet owners name their pets a lot. ITs just- a thing. We all do it.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
 
image

</details>

## Changelog

:cl:
add: Pet owners can now name and describe their pet
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
drafted pending testing

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 1]
```diff
@@ -28,6 +28,12 @@
 
 	var/obj/item/pet_carrier/carrier = new /obj/item/pet_carrier(get_turf(quirk_holder))
 	var/mob/living/simple_animal/pet/pet = new pet_type(carrier)
+	var/new_name = client_source?.prefs.read_preference(/datum/preference/text/pet_name)
+	if (new_name)
+		pet.name = new_name
+	var/new_desc = client_source?.prefs.read_preference(/datum/preference/text/pet_desc)
+	if (new_desc)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/pet_owner/pet_owner.dm:35]: You might want to trim whitespaces from the start and end of the names and descriptions, to also make sure that they're not just whitespace, so we don't have people naming their pet ` `, y'know. It'll look nicer, and should mean that the check will work as long as they have actual characters in it.

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 2]
```diff
@@ -93,3 +99,33 @@ GLOBAL_LIST_INIT(possible_player_pet, list(
 
 /datum/preference/choiced/pet_owner/apply_to_human(mob/living/carbon/human/target, value)
 	return
+
+/datum/preference/text/pet_name
+	category = PREFERENCE_CATEGORY_MANUALLY_RENDERED
+	savefile_key = "pet_name"
+	savefile_identifier = PREFERENCE_CHARACTER
+	can_randomize = FALSE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/pet_owner/pet_owner.dm:107]: Just so people don't exaggerate with the size of their pets' names.
```suggestion
	can_randomize = FALSE
	maximum_value_length = 32
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/pet_owner/pet_owner.dm (hunk 1 above)
Comment (nikothedude) [reply modular_skyrat/modules/pet_owner/pet_owner.dm:35]: regex :)

**Comment by nikothedude** [comment]
i am moving to galaxia and am focusing my efforts there

**Pull request closed without merging**